File: Projects/sentimental-readability/readability.py
from cs50 import get_string


def main():

    text = get_string("Text: ")
    letters = get_letters(text)
    words = get_words(text)
    sentences = get_sentences(text)
    grade = get_grade(letters, words, sentences)
    print_grade(grade)


def get_letters(text):
    letters = 0
    for char in text:
        if char.isalpha() == True:
            letters += 1

    return letters


def get_words(text):
    words = 0
    for char in text:
        if char == " ":
            words += 1
    words += 1

    return words


def get_sentences(text):
    sentences = 0
    for char in text:
        if char == "." or char == "!" or char == "?":
            sentences += 1

    return sentences


def get_grade(letters, words, sentences):
    l = letters / words * 100
    s = sentences / words * 100
    grade = 0.0588 * l - 0.296 * s - 15.8
    grade = round(grade)

    return grade


def print_grade(grade):
    if grade >= 16:
        print("Grade 16+")
    elif grade < 1:
        print("Before Grade 1")
    else:
        print(f"Grade {grade}")


if __name__ == "__main__":
    main()

File: Projects/scrabble/readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

float count_letters(string);
float count_words(string);
float count_sent(string);
void get_grade(int);

int main(void)
{
    string text = get_string("Text: ");
    float words = count_words(text);
    float letters = count_letters(text);
    float sent = count_sent(text);

    //compute Coleman_Liau index
    float L = letters / words * 100;
    float S = sent / words * 100;
    float grade = 0.0588 * L - 0.296 * S - 15.8;
    int final_grade = round(grade);//round to nearest integer

    //print final grade level
    get_grade(final_grade);


}

//count the letters in the text
float count_letters(string t)
{
    int len = strlen(t);
    int count = 0;

    for (int i = 0; i < len; i++)
    {
        if ((t[i] >= 'A' && t[i] <= 'Z') || (t[i] >= 'a' && t[i] <= 'z'))
        {
            count++;
        }
    }
    return count;
}

//count the words in the text
float count_words(string t)
{
    int len = strlen(t);
    int count = 0;

    for (int i = 0; i < len; i++)
    {
        if (t[i] == 32)
        {
            count++;
        }
    }
    return count + 1;
}

//count the sentences in the text
float count_sent(string t)
{
    int len = strlen(t);
    int count = 0;

    for (int i = 0; i < len; i++)
    {
        if (t[i] == '.' || t[i] == '?' || t[i] == '!')
        {
            count++;
        }
    }
    return count;
}

//print the grade level
void get_grade(int n)
{
    if (n >= 1 && n <= 15)
    {
        printf("Grade %i\n", n);
    }
    else if (n < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade 16+\n");
    }
}